src/CheckedFile.cpp

```diff
@@ -565,6 +631,14 @@ void CheckedFile::close()
 
       fd_ = -1;
    }
+
+   if (bufView_ != nullptr) {
+      delete bufView_;
+      bufView_ = nullptr;
+
+      // WARNING: do NOT delete buffer of streamView_ because

```

Comment:
`streamView_`? Do you mean `bufView_`?

If so, this comment should be in the BufferView class, not here. Maybe just add a comment on the BufferView constructor that the caller retains ownership of the pointer.